File: Algorithms/01-Warmup/04-diagonal-difference/04.py
#!/usr/bin/env python


def diagonalDifference(a):
    sum_from_left = 0
    sum_from_right = 0

    for i in range(0, len(a)):
        sum_from_left += a[i][0 + i]
        sum_from_right += a[i][- (i - 1)]
    # Complete this function
    total = abs(sum_from_left - sum_from_right)
    return total


if __name__ == "__main__":
    n = int(input().strip())
    a = []
    for a_i in range(n):
       a_t = [int(a_temp) for a_temp in input().strip().split(' ')]
       a.append(a_t)
    result = diagonalDifference(a)
    print(result)

File: Algorithms/01-Warmup/04-diagonal-difference/04.rb
#!/usr/bin/env ruby

def diagonal_difference(a)
    sum_from_left = 0
    sum_from_right = 0

    (0...a.length).each do |i|
        sum_from_left += a[i][0 + i]
        sum_from_right += a[i][-(i + 1)]
    end

    total = sum_from_left - sum_from_right
    total.abs
end

n = gets.strip.to_i
a = Array.new(n)
(0...n).each do |a_i|
    a_t = gets.strip
    a[a_i] = a_t.split(" ").map(&:to_i)
end
result = diagonal_difference(a)
puts result
